From: SF2000 Linux contributors
Subject: [PATCH] toolchain: add GCC 16.1, binutils 2.46.1 and uClibc-ng 1.0.58

Keep the SF2000 toolchain on the newest stable upstream releases.  Buildroot
2026.05.1 predates these point releases, so carry the small package metadata
update locally rather than using an unversioned Buildroot snapshot.

---
 package/binutils/Config.in.host | 4 ++--
 package/binutils/binutils.hash  | 1 +
 package/gcc/Config.in.host      | 7 +++++++
 package/gcc/gcc.hash            | 2 ++
 package/uclibc/uclibc.hash      | 1 +
 package/uclibc/uclibc.mk        | 2 +-
 6 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/package/binutils/Config.in.host b/package/binutils/Config.in.host
--- a/package/binutils/Config.in.host
+++ b/package/binutils/Config.in.host
@@ -25 +25 @@ config BR2_BINUTILS_VERSION_2_46_X
-	bool "binutils 2.46.0"
+	bool "binutils 2.46.1"
@@ -35 +35 @@ config BR2_BINUTILS_VERSION
-	default "2.46.0"	if BR2_BINUTILS_VERSION_2_46_X
+	default "2.46.1"	if BR2_BINUTILS_VERSION_2_46_X
diff --git a/package/binutils/binutils.hash b/package/binutils/binutils.hash
--- a/package/binutils/binutils.hash
+++ b/package/binutils/binutils.hash
@@ -3,2 +3,3 @@ sha512  b85d3bbc0e334cf67a96219d3c7c65fbf3e832b2c98a7417bf131f3645a0307057ec81cd2b29ff2563cec53e3d42f73e2c60cc5708e80d4a730efdcc6ae14ad7  binutils-2.44.tar.xz
 sha512  ea030419eba387579ab717be7e3223fc99e93b586860b06003c12489f93441640d4082736f76aa5e98233db4f46e232f536a45e471486de1f5b64e1b827c167e  binutils-2.45.1.tar.xz
 sha512  32f880bb4f69351f4ae54a5d00359625c6c49d8e76624fb5cffdf174c79c8d3212f66225b81c12933c6ed59604ab652560773dd92fab384b930c97a9d4e1fdf2  binutils-2.46.0.tar.xz
+sha512  a5c65e56e400ed3fb8906a995dbb93eb5bea54b16344244653d7f44ef29ceb60270da263b19d25302c37759784e14fcb4b9421b29e0e2c7f450bd99f6bb4595c  binutils-2.46.1.tar.xz
diff --git a/package/gcc/Config.in.host b/package/gcc/Config.in.host
--- a/package/gcc/Config.in.host
+++ b/package/gcc/Config.in.host
@@ -47,4 +47,9 @@ config BR2_GCC_VERSION_15_X
 	depends on !BR2_POWERPC_CPU_HAS_SPE
 	select BR2_TOOLCHAIN_GCC_AT_LEAST_15
 
+config BR2_GCC_VERSION_16_X
+	bool "gcc 16.x"
+	depends on !BR2_POWERPC_CPU_HAS_SPE
+	select BR2_TOOLCHAIN_GCC_AT_LEAST_16
+
 endchoice
@@ -71,4 +76,5 @@ config BR2_GCC_VERSION
 	default "13.4.0"    if BR2_GCC_VERSION_13_X
 	default "14.4.0"    if BR2_GCC_VERSION_14_X
 	default "15.3.0"    if BR2_GCC_VERSION_15_X
+	default "16.1.0"    if BR2_GCC_VERSION_16_X
 	default "arc-2024.12-release" if BR2_GCC_VERSION_ARC
diff --git a/package/gcc/gcc.hash b/package/gcc/gcc.hash
--- a/package/gcc/gcc.hash
+++ b/package/gcc/gcc.hash
@@ -7,2 +7,4 @@ sha512  725ed8bdd43ef1726ffe8b5e8615a13e247fac9575b7626ae013a2975d000ea213212dc414b2f2631ac4785c1c8beca85555222faf9904d3b2fa6a3807a83a15  gcc-14.4.0.tar.xz
 # From https://gcc.gnu.org/pub/gcc/releases/gcc-15.3.0/sha512.sum
 sha512  0de9e296153b52c021b1c7e63c9c62151d7a0ac03f23ce6e9f772c1b0eb783f6acdd81cc4567bfe4128a6f64968c2cfc8eff40b36229cba7425349f7d637c654  gcc-15.3.0.tar.xz
+# Locally calculated from the upstream release archive
+sha512  b3454958891ab47e1e5b6cb9396c0ad3b04f32fe2a7bf1153a143f21013fdb6b295ca94c98964698a688e4c1d7555ffd8ffbc20187507cce6b1c32cbcc09897a  gcc-16.1.0.tar.xz
diff --git a/package/uclibc/uclibc.hash b/package/uclibc/uclibc.hash
--- a/package/uclibc/uclibc.hash
+++ b/package/uclibc/uclibc.hash
@@ -1,2 +1,3 @@
 # From https://downloads.uclibc-ng.org/releases/1.0.57/uClibc-ng-1.0.57.tar.xz.sha256
 sha256  8bc734b584e23ff6ae3d0ebb4c0fb1d1d814c58c82822b93130d436afa7ace8b  uClibc-ng-1.0.57.tar.xz
+sha256  9e8100a442f7079b9728ca286e3035f67df0578606166ea2d1ef865a4a12ddb9  uClibc-ng-1.0.58.tar.xz
diff --git a/package/uclibc/uclibc.mk b/package/uclibc/uclibc.mk
--- a/package/uclibc/uclibc.mk
+++ b/package/uclibc/uclibc.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-UCLIBC_VERSION = 1.0.57
+UCLIBC_VERSION = 1.0.58
 UCLIBC_SOURCE = uClibc-ng-$(UCLIBC_VERSION).tar.xz
 UCLIBC_SITE = https://downloads.uclibc-ng.org/releases/$(UCLIBC_VERSION)
